asr-python-advanced-nemo-ngram-training-and-finetuning.ipynb

```diff
@@ -0,0 +1,3133 @@
+{
+  "cells": [
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ToDOY7a1NpJp"
+      },
+      "source": [
+        "<img src=\"http://developer.download.nvidia.com/notebooks/dlsw-notebooks/riva_asr_asr-python-advanced-customize-vocabulary-and-lexicon/nvidia_logo.png\" style=\"width: 90px; float: right;\">\n",
+        "\n",
+        "<a id='isc-task-description'></a>\n",
+        "# Language Modeling\n",
+        "\n",
+        "### Task Description\n",
+        "\n",
+        "Language modeling returns a probability distribution over a sequence of words. Besides assigning a probability to a sequence of words, the language models also assign a probability for the likelihood of a given word (or a sequence of words) that follows a sequence of words. <br>\n",
+        "\n",
+        "> The sentence:  **all of a sudden I notice three guys standing on the sidewalk**\n",
+        "> would be scored higher than \n",
+        "> the sentence: **on guys all I of notice sidewalk three a sudden standing the** by the language model. <br>\n",
+        "\n",
+        "A language model trained on large corpus can significantly improve the accuracy of an ASR system as suggested in recent research."
+      ]
+    },
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "BqDDGrWHN4YK"
+      },
+      "source": [
+        "### N-gram Language Model\n",
+        "There are primarily two types of language models:\n",
+        "\n",
+        "- **N-gram language models**: These models use frequency of n-grams to learn the probability distribution over words. Two benefits of N-gram language models are simplicity and scalability – with a larger `n`, a model can store more context with a well-understood space–time tradeoff, enabling small experiments to scale up efficiently.\n",
+        "- **Neural language models**: These models use different kinds of neural networks to model the probability distribution over words, and have surpassed the N-gram language models in the ability to model language, but are generally slower to evaluate.\n",
```

Comment:
What do you mean by `slower to evaluate` ? Did you mean slower inference?

Reply:
Yes